cl_manycore/libraries/bsg_manycore_eva.cpp

```diff
@@ -284,6 +296,91 @@ static uint32_t default_get_dram_x_shift(const hb_mc_manycore_t *mc)
         }
 }
 
+static uint32_t default_get_dram_x_shift_dep(const hb_mc_manycore_t *mc)
+{
+        const hb_mc_config_t *cfg = hb_mc_manycore_get_config(mc);
+        if (hb_mc_manycore_dram_is_enabled(mc)) {
+                return hb_mc_config_get_vcache_bitwidth_data_addr(cfg);
+        } else {
+                return ceil(log2(hb_mc_config_get_vcache_size(cfg))); // clog2(victim cache size)
+        }
+}
+
+// See comments on default_eva_to_npa_dram 
+static int default_eva_get_x_coord_dram(const hb_mc_manycore_t *mc,
+                                        const hb_mc_config_t *cfg,
+                                        const hb_mc_eva_t *eva,
+                                        hb_mc_idx_t *x) { 
+        uint32_t stripe_log = default_get_dram_stripe_size_log(mc);
+        uint32_t xmask = default_get_dram_x_bitidx(cfg);
+        uint32_t dram_max_x_coord = default_get_dram_max_x_coord(cfg);
+
+        *x = (hb_mc_eva_addr(eva) >> stripe_log) & xmask;
+        if ( *x > dram_max_x_coord) { 
+                bsg_pr_err("%s: Translation of EVA 0x%08" PRIx32 " failed. The X coordinate "
+                           "of the DRAM bank for the requested EPA %d is larger than max %d\n.",
+                           __func__, hb_mc_eva_addr(eva),
+                           *x, dram_max_x_coord);
+                return HB_MC_INVALID;
+        }
+        return HB_MC_SUCCESS;
+}
+
+// See comments on default_eva_to_npa_dram 
+static int default_eva_get_y_coord_dram(const hb_mc_manycore_t *mc,
+                                        const hb_mc_config_t *cfg,
+                                        const hb_mc_eva_t *eva,
+                                        hb_mc_idx_t *y) { 
+        // Y dimension is fixed at the bottom column
+        *y = hb_mc_config_get_dram_y(cfg);
+        return HB_MC_SUCCESS;
+}
+
+// See comments on default_eva_to_npa_dram 
+static int default_eva_get_epa_dram (const hb_mc_manycore_t *mc,
+                                     const hb_mc_config_t *cfg,
+                                     const hb_mc_eva_t *eva,
+                                     hb_mc_epa_t *epa,
+                                     size_t *sz) { 
+ 
+        uint32_t xdimlog    = default_get_x_dimlog(cfg);
+        uint32_t stripe_log = default_get_dram_stripe_size_log(mc);
+	uint32_t shift      = stripe_log + xdimlog;
+        
+        *epa = (  (hb_mc_eva_addr(eva) & MAKE_MASK(stripe_log)) 
```

Comment:
I'd prefer this line of code were not a one liner. Maybe even add some comments.

Reply:
OK I changed it a little and added some comments. How does it look now?